corehq/apps/hqmedia/models.py

```diff
@@ -540,10 +552,198 @@ def _log_media_deletion(app, deleted_media):
+                                                       is_menu_media=True, **kwargs))
+
+            # Print template
+            if display and details.display == 'long' and details.print_template:
+                media.append(ApplicationMediaReference(details.print_template['path'],
+                                                       media_class=CommCareMultimedia, **kwargs))
+
+            # Icon-formatted columns
+            for column in details.get_columns():
+                if column.format == 'enum-image':
+                    for map_item in column.enum:
+                        icons = list(map_item.value.values())
+                        media.extend([ApplicationMediaReference(icon, media_class=CommCareImage,
+                                                                is_menu_media=True, **kwargs)
+                                      for icon in icons if icon])
+
+        return media
+
+    def rename_media(self, old_path, new_path):
+        count = 0
+
+        count += self.rename_menu_media(self, old_path, new_path)
+
+        if self.case_list_form.form_id:
+            count += self.rename_menu_media(self.case_list_form, old_path, new_path)
+
+        if hasattr(self, 'case_list') and self.case_list.show:
+            count += self.rename_menu_media(self.case_list, old_path, new_path)
+
+        for name, details, display in self.get_details():
+            # Case list lookup
+            if display and details.display == 'short' and details.lookup_enabled and details.lookup_image:
+                if details.lookup_image == old_path:
+                    details.lookup_image = new_path
+                    count += 1
+
+            # Print template
+            if display and details.display == 'long' and details.print_template:
+                details.print_template['path'] = new_path
+                count += 1
+
+            # Icon-formatted columns
+            for column in details.get_columns():
+                if column.format == 'enum-image':
+                    for map_item in column.enum:
+                        for lang, icon in six.iteritems(map_item.value):
+                            if icon == old_path:
+                                map_item.value[lang] = new_path
+                                count += 1
+
+        return count
+
+
+class FormMediaMixin(MediaMixin):
+    def get_media_ref_kwargs(self):
+        module = self.get_module()
+        return {
+            'app_lang': module.get_app().default_language,
+            'module_name': module.name,
+            'module_id': module.id,
+            'form_name': self.name,
+            'form_id': self.unique_id,
+            'form_order': self.id,
+        }
+
+    @memoized
+    def memoized_xform(self):
+        return self.wrapped_xform()
+
+    def all_media(self):
+        kwargs = self.get_media_ref_kwargs()
+        media = []
+
+        media.extend(self.menu_media(self))
```

Comment:
whats the reason for using extend and not just a straight assignment?

Reply:
Not deliberate. Updated.